ground/gcs/src/plugins/config/outputchannelform.cpp

```diff
@@ -386,15 +386,30 @@ void OutputChannelForm::updateMaxSpinboxValue(UAVObject *obj)
         foreach(qint32 channel, channelBank) {
             // ... and if there's a match, set the maximum values and return
             if (channel-1 == m_index) {
-                if (actuatorSettingsData.TimerUpdateFreq[i] == 0)
-                    return;
-
-                double maxPulseWidth = round(10000000.0 / actuatorSettingsData.TimerUpdateFreq[i]);
-
-                // Saturate at the UAVO's maximum value
-                if (maxPulseWidth > std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max())
-                    maxPulseWidth = std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max();
-
+                
+                double maxPulseWidth = 0;
+                
+                if (actuatorSettingsData.TimerUpdateFreq[i] != 0)
+                {
+                    double pwm_freq = 0;
+                    if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_1MHZ)
+                        pwm_freq = 1000000;
+                    else if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_12MHZ)
+                        pwm_freq = 12000000;
+                    maxPulseWidth = round(pwm_freq / actuatorSettingsData.TimerUpdateFreq[i]);
+                    
+                    // Saturate at the UAVO's maximum value
+                    if (maxPulseWidth > std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max())
+                        maxPulseWidth = std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max();
+                } else {
+                    // SyncPWM has been selected, the actual pulse limit is a function of sensor rate.
+                    // Simply limit to 1ms here to account for 1000Hz sensor rate.
+                    if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_1MHZ)
+                        maxPulseWidth = 1000;
+                    else if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_12MHZ)
+                        maxPulseWidth = 12000;
```

Comment:
The maxPulseWidth = 12000µs = 12ms in higher resolution? This can't be right. Why do want to limit the pulse width in sync mode. The pulse width is given in µs independent from the base frequency. If you want to limit anything, you have to look at the sensor rate, but why? Limiting the pulse width to 1ms makes normal ESCs (without OneShot) unusable in synced mode. 
BTW I wouldn't use round() here better choose ceil().